src/App.tsx

```diff
@@ -1,26 +1,316 @@
+      todoField.current.focus();
+    }
+  }, [todosLength, errorMessageAddTodo]);
+
+  const handleFilterChange = (status: FilterStatus) => {
+    setFilterStatus(status);
+  };
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterStatus) {
+      case 'active':
+        return !todo.completed;
+
+      case 'completed':
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
+  const addTodo = async (newTitle: string) => {
+    setTempTodo({ id: 0, title: newTitle, completed: false, userId: USER_ID });
+    setIsLoading('adding');
+    setDisabledInput(true);
+
+    return createTodo({
+      title: newTitle.trim(),
+      completed: false,
+      userId: USER_ID,
+    })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTempTodo(null);
+      })
+      .catch(error => {
+        setTempTodo(null);
+
+        if (error instanceof Error) {
+          throw new Error(ERROR_ADD_TODO);
+        }
+
+        setErrorMessage((error as Error).message);
+        throw error;
+      })
+      .finally(() => {
+        resetError();
+        setIsLoading(false);
+        setDisabledInput(false);
+      });
+  };
+
+  const renameTodo = async (id: number, newTitle: string): Promise<void> => {
+    const currentTodo = todos.find(todo => todo.id === id);
+
+    if (currentTodo && currentTodo.title.trim() === newTitle.trim()) {
+      return Promise.resolve();
+    }
+
+    setIsLoading('updating');
+
+    try {
+      const updatedTodo = await updateTodo(id, newTitle.trim());
+
+      setTodos(prevTodos =>
+        prevTodos.map(todo =>
+          todo.id === updatedTodo.id ? updatedTodo : todo,
+        ),
+      );
+    } catch (error) {
+      setErrorMessage(ERROR_UPDATE_TODO);
+      throw error;
+    } finally {
+      resetError();
+      setIsLoading(false);
+    }
+  };
+
+  const toggleTodo = async (id: number) => {
+    setIsLoading('toggling');
+
+    try {
+      const todo = todos.find(t => t.id === id);
+      const updatedTodo = await toggleUpdateTodo(id, !todo?.completed);
+
+      setTodos(prevTodos =>
+        prevTodos.map(prevTodo =>
+          prevTodo.id === updatedTodo.id ? updatedTodo : prevTodo,
+        ),
+      );
+    } catch (error) {
+      setErrorMessage(ERROR_UPDATE_TODO);
+    } finally {
+      resetError();
+      setIsLoading(false);
+    }
+  };
+
+  const removeTodo = async (id: number) => {
+    setIsLoading('deleting');
+
+    return deleteTodos(id)
+      .then(() => setTodos(todos.filter(todo => todo.id !== id)))
+      .catch(() => {
+        setTodos(todos);
+        setErrorMessage(ERROR_DELETE_TODO);
+      })
+      .finally(() => {
+        resetError();
+        setIsLoading(false);
+      });
+  };
+
+  const handleClearCompleted = async () => {
+    const completedTodosIds = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    setIsLoading('completed');
```

Comment:
```suggestion
    setIsLoading(true);
```
just use boolean values